synapse/http/servlet.py

```diff
@@ -126,14 +134,22 @@ def parse_string(request, name, default=None, required=False,
             is not one of those allowed values.
     """
     return parse_string_from_args(
-        request.args, name, default, required, allowed_values, param_type,
+        request.args, name, default, required, allowed_values, param_type, encoding
     )
 
 
 def parse_string_from_args(args, name, default=None, required=False,
-                           allowed_values=None, param_type="string"):
+                           allowed_values=None, param_type="string", encoding='ascii'):
+
+    if not isinstance(name, bytes) and encoding:
```

Comment:
it feels odd that `encoding` applies to both the `name` and the result. I feel like it should just apply to the result, and the name should *always* be encoded as ascii (if it's not already a `bytes`)

Reply:
Yeah, fair. Fixed.